src/main/java/org/rumbledb/runtime/postfix/ArrayUnboxingIterator.java

```diff
@@ -114,4 +121,35 @@ private void setNextResult() {
         JavaRDD<Item> resultRDD = childRDD.flatMap(transformation);
         return resultRDD;
     }
+
+    @Override
+    public boolean implementsDataFrames() {
+        return true;
+    }
+
+    public Dataset<Row> getDataFrame(DynamicContext context) {
+        Dataset<Row> childDataFrame = this.children.get(0).getDataFrame(context);
+        String key;
+        childDataFrame.createOrReplaceTempView("array");
+        StructType schema = childDataFrame.schema();
+        String[] fieldNames = schema.fieldNames();
+        if (
+            fieldNames.length == 1 && Arrays.asList(fieldNames).contains(SparkSessionManager.atomicJSONiqItemColumnName)
+        ) {
+            int i = schema.fieldIndex(SparkSessionManager.atomicJSONiqItemColumnName);
+            StructField field = schema.fields()[i];
+            DataType type = field.dataType();
+            if (type instanceof ArrayType) {
+                return childDataFrame.sparkSession()
+                    .sql(
+                        String.format(
+                            "SELECT explode(`%s`) AS `%s` FROM array",
+                            SparkSessionManager.atomicJSONiqItemColumnName,
+                            SparkSessionManager.atomicJSONiqItemColumnName
+                        )
+                    );
+            }
+        }
+        return childDataFrame.sparkSession().sql("SELECT * FROM array WHERE false");
```

Comment:
This line to return an empty result will be used for many similar function calls I think. Would it make sense to have a static empty DataFrame, and return this from all similar places instead of making this sql() call? I feel that this could go a long way for readability.

Reply:
Very good catch! This is fixed.

Reply:
Looks very clean now. Thanks